### ISSUE TITLE
Match team_keys lists include "B" team key instead of the actual team number they are playing as

### ISSUE THREAD

**Issue by eforbes** [the-blue-alliance/the-blue-alliance#4808]
## Expected Behavior
GET `/api/v3/event/2022cacg/simple` should return valid team keys that match the number the robot is operating as at the event

## Current Behavior
GET `/api/v3/event/2022cacg/matches/simple`

<pre>
[
  {
    "actual_time": null,
    "alliances": {
      "blue": {
        "dq_team_keys": [],
        "score": 80,
        "surrogate_team_keys": [],
        "team_keys": [
          "frc5419",
          "frc670",
          "frc841"
        ]
      },
      "red": {
        "dq_team_keys": [],
        "score": 58,
        "surrogate_team_keys": [],
        "team_keys": [
          "frc8",
          "frc2813",
          "<b>frc766B</b>"
        ]
      }
    },
    "comp_level": "f",
    "event_key": "2022cacg",
    "key": "2022cacg_f1m1",
    "match_number": 1,
    "predicted_time": null,
    "set_number": 1,
    "time": 1665352513,
    "winning_alliance": "blue"
  },

...
</pre>

## Possible Solution
Return the actual team numbers (9XXX) in the API instead of the "B" team numbers. Alternatively, return both or add a parameter so consumers of the API can choose which type of team number to use depending on their use case.

## Steps to Reproduce (for bugs)
1. GET `/api/v3/event/2022cacg/matches/simple`

## Context
With the current implementation of the API, I don't see any way to reconcile the team numbers from the actual event and the numbers returned by the API. I understand the historical value in representing B robots as part of their real team, but since they actually operate at events under the 9XXX numbers (have that number on their bumpers, are announced by the emcee as that etc) it makes it more difficult to use this API in live contexts.

The only workaround I've come up with is maintaining my own mapping of 9XXX team number to "B" robot number, but it would have to be constructed manually for each offseason using B teams (with this information only available from event management or being at the event (?)).

## Your Environment
* Version used: API v3

**Comment by fangeugene** [comment]
Hmm this is an interesting problem. I understand why it would make sense for TBA to show (both in the API and on the website/app) the 9XXX team number during an event, but it doesn't make sense to keep those around after the event since it can lead to long-term issues where the 9XXX team number collides with a real team.

Maybe we could switch from 9XXX to XXXB after the event ends, but that's still not great because you end up with a short period where you could still have team number collisions. And the infra isn't built to do that easily. Alternatively we could provide the B team mappings as part of the API?

cc@phil-lopreiato @ZachOrr @FletcherS7 any thoughts?

**Comment by phil-lopreiato** [comment]
Yeah, this is a tricky one.

I think providing the B team mappings in the API is probably the best approach here?

**Comment by bdaroz** [comment]
I think we've always said that the info returned represents the team at the event. While the team may have their normal number +`99xx` for additional robots, a change to the data historically returned could have unintended consequences, especially we are getting new teams in the 96xx range now.

I'm all for including a mapping function like:
```
{
     ...     
     teamMapping: [
          {"frc5881B:"frc9997"},{"frc254B":"frc9998"}
     ]
     .....
}
```

**Comment by eforbes** [comment]
I would use the mappings. Adds a bit of complexity on the API consumer side but it will solve my problem for sure.

Thanks!

**Comment by lethosor** [comment]
I had experimented with B teams in the API in https://github.com/the-blue-alliance/the-blue-alliance/pull/2569, but it requires changing a lot of current constraints.

I think exposing the team mapping like that is probably a simpler way to go.

**Comment by fangeugene** [comment]
Potential solution: https://www.chiefdelphi.com/t/the-blue-alliance-2025-site-updates-features/495235/29?u=eugene_fang

**Comment by lethosor** [comment]
Yeah, I've been thinking that would be a good approach to take as well. The API would just need to expose `remap_teams` to clients.

It would mean changing all frontends, but also would resolve some of the data inconsistencies like the one in that thread and the ones my PR was attempting to resolve.

**Comment by eforbes** [comment]
#8028 fixes my problem, thanks!
